EventListener/ThemeControllerListener.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Liip\ThemeBundle\EventListener;
+
+use Liip\ThemeBundle\ActiveTheme;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+
+/**
+ * Listener to rewrite controllers when configured.
+ *
+ * This can make sense e.g. when using themes to do A/B testing.
+ *
+ * @author David Buchmann <david@liip.ch>
+ */
+class ThemeControllerListener
+{
+    /**
+     * @var ActiveTheme Information about the active theme
+     */
+    private $activeTheme;
+
+    const ATTRIBUTE_KEY = 'theme_controllers';
+
+    /**
+     * @param ActiveTheme $activeTheme to know which theme is active.
+     */
+    public function __construct(ActiveTheme $activeTheme)
+    {
+        $this->activeTheme = $activeTheme;
+    }
+
+    /**
+     * Kick in right after routing and check if the route defines options to
+     * use a different controller for the new theme.
+     *
+     * @param GetResponseEvent $event
+     */
+    public function onKernelRequest(GetResponseEvent $event)
+    {
+        $request = $event->getRequest();
+        // TODO what if we are on the default?
+        $theme = $this->activeTheme->getName();
```

Comment:
okay. is the isset below a problem if we do isset($controllers[null])? should i do if ($theme && isset()) instead or is it good like this?